app/shop.py

```diff
@@ -0,0 +1,45 @@
+from app.customer import Customer
+from datetime import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int],
+            products: dict[str, float]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def products_cost(self, customer: Customer) -> float:
+        product_cart = customer.product_cart
+        product_cost = 0
+        for product, qty in product_cart.items():
+            product_cost += qty * self.products.get(product)

```

Comment:
The expression `self.products.get(product)` may return `None` if the product is not found in the shop's inventory. This could lead to a `TypeError` when multiplying `None` by `qty`. Consider providing a default value (e.g., 0) to avoid this issue.